### PR TITLE
[MM-872]: Added server testcases for api/post_action.go

### PR DESCRIPTION
**Pull request by Kshitij-Katiyar**
#### Summary
Added server testcases for api/post_action.go

### CODE REVIEW

**Review by wiggin77 — APPROVED** [review]
LGTM :+1:

File: calendar/api/post_action_test.go [hunk 1]
```diff
@@ -0,0 +1,619 @@
...
+			},
+		},
+		{
+			name: "Invalid request body",
+			setup: func(req *http.Request) {
+				req.Header.Set(MMUserIDHeader, MockUserID)
+				req.Body = io.NopCloser(bytes.NewBufferString("invalid json"))
+			},
+			assertions: func(rec *httptest.ResponseRecorder, mscal engine.Engine, user *engine.User, eventID, option, postID string) {
+				assert.Equal(t, http.StatusOK, rec.Result().StatusCode)
```
Comment (raghavaggarwal2308) [inline calendar/api/post_action_test.go:51]: Why are we getting status OK in all the cases. Please check if we need to update our code to return proper status codes

File: calendar/api/post_action_test.go [hunk 2]
```diff
@@ -0,0 +1,619 @@
...
+					Context: map[string]interface{}{
+						"value":     true,
+						"change_to": "away",
+					},
+				}
+				bodyBytes, _ := json.Marshal(requestBody)
+				req.Body = io.NopCloser(bytes.NewBuffer(bodyBytes))
+			},
+			assertions: func(rec *httptest.ResponseRecorder) {
+				assert.Equal(t, http.StatusOK, rec.Result().StatusCode)
```
Comment (raghavaggarwal2308) [inline calendar/api/post_action_test.go:539]: Same here

File: calendar/api/test_utils.go [hunk 3]
```diff
@@ -39,6 +41,64 @@ func (m *MockNotificationProcessor) Enqueue(notifications ...*remote.Notificatio
 func (m *MockNotificationProcessor) Configure(env engine.Env) {}
 func (m *MockNotificationProcessor) Quit()                    {}
 
+type mockClient struct{}
```
Comment (raghavaggarwal2308) [inline calendar/api/test_utils.go:44]: Can't we use mockgen for this? OR this coming from an external library?

**Review by raghavaggarwal2308 — CHANGES_REQUESTED** [review]

File: calendar/api/post_action_test.go (hunk 1 above)
Comment (Kshitij-Katiyar) [reply calendar/api/post_action_test.go:51]: We are not writing any status code in the whole file. I think we should make it a separate task to return proper status code for all the API in the plugin

Comment (raghavaggarwal2308) [reply calendar/api/post_action_test.go:51]: @Kshitij-Katiyar Sure, please log a issue for the same

**Review by raghavaggarwal2308 — APPROVED** [review]

**Pull request merged**